### PR TITLE
Remove very verbose log line with the encoded trace

### PR DESCRIPTION
**Pull request by albertvaka**
### What does this PR do?
Hides a log line that is very spammy and IMHO not useful behind the `DD_TRACE_ENCODING_DEBUG` env var.

### Motivation
If `DD_TRACE_DEBUG` was set, we printed each trace after encoding as hex. This was a very long but not very readable log line, and probably not useful anymore.

### CODE REVIEW

**Review by rochdev — CHANGES_REQUESTED** [review]
Let's move it under a new env var instead so that it's still available if needed.

File: packages/dd-trace/src/encode/0.4.js [hunk 1]
```diff
@@ -13,6 +15,11 @@ float64Array[0] = -1
 
 const bigEndian = uInt8Float64Array[7] === 0
 
+const debugEncoding = isTrue(coalesce(
+  process.env.DD_TRACE_ENCODING_DEBUG,
+  false
+))
```
Comment (Qard) [inline packages/dd-trace/src/encode/0.4.js:21]: This should probably go in the Config class and get passed into AgentEncoder at construction.

Comment (albertvaka) [reply packages/dd-trace/src/encode/0.4.js:21]: 👍 Should I create a new Config instance or pass an existing one? If the second, where do I get the existing instance from?

Comment (Qard) [reply packages/dd-trace/src/encode/0.4.js:21]: Yes, something like that.

@rochdev What do you think is the best way to handle this?

Comment (albertvaka) [reply packages/dd-trace/src/encode/0.4.js:21]: Actually this change breaks lots of tests that have assert on arguments passed to things. I don't think we want to make such a big change.

**Review by rochdev — COMMENTED** [review]
This looks fine in general, but it's a lot of changes. I think for debugging env vars maybe they could be used directly and not go through the config object? These wouldn't need to be accessed across the codebase anyway, so passing them around just creates unnecessary noise.

File: packages/dd-trace/src/encode/0.4.js (hunk 1 above)
Comment (rochdev) [reply packages/dd-trace/src/encode/0.4.js]: I missed this comment but I replied as a review.

**Review by rochdev — APPROVED** [review]

**Pull request merged**